x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -471,11 +477,7 @@ protected Map<String, EnrichPolicy> availablePolicies() {
         return projectResolver.getProjectMetadata(clusterService.state()).custom(EnrichMetadata.TYPE, EnrichMetadata.EMPTY).getPolicies();
     }
 
-    protected void getRemoteConnection(String cluster, ActionListener<Transport.Connection> listener) {
-        remoteClusterService.maybeEnsureConnectedAndGetConnection(
-            cluster,
-            remoteClusterService.isSkipUnavailable(cluster).orElse(true) == false,
-            listener
-        );
+    protected void getRemoteConnection(String cluster, EsqlExecutionInfo executionInfo, ActionListener<Transport.Connection> listener) {
```

Comment:
Not saying this is wrong, but here you don't actually need `EsqlExecutionInfo`, only one flag that describes `executionInfo.shouldSkipOnFailure(cluster) == false` which is `ensureConnected` value. So maybe this should be `boolean ensureConnected` too?